sql-plugin/src/main/scala/com/nvidia/spark/rapids/limit.scala

```diff
@@ -37,6 +37,97 @@ import org.apache.spark.sql.execution.{CollectLimitExec, LimitExec, SparkPlan}
 import org.apache.spark.sql.execution.exchange.ENSURE_REQUIREMENTS
 import org.apache.spark.sql.vectorized.{ColumnarBatch, ColumnVector}
 
+object GpuBaseLimitExec {
+
+  private def sliceBatchAndCloseWithRetry(
+      spillBatch: SpillableColumnarBatch,
+      start: Int,
+      length: Int): ColumnarBatch = {
+    val end = Math.min(start + length, spillBatch.numRows())
+    RmmRapidsRetryIterator.withRetryNoSplit(spillBatch) { _ =>
+      withResource(spillBatch.getColumnarBatch()) { batch =>
+        val subCols = (0 until batch.numCols()).safeMap { i =>
+          val col = batch.column(i).asInstanceOf[GpuColumnVector]
+          val subVector = col.getBase.subVector(start, end)
+          assert(subVector != null)
+          GpuColumnVector.from(subVector, col.dataType)
+        }
+        new ColumnarBatch(subCols.toArray, end - start)
+      }
+    }
+  }
+
+  def apply(
```

Comment:
nit: could we call this something different. To me GpuBaseLimitExec.apply should return a GpuBaseLimitExec, we are not 100% consistent about that.  Could we rename the object to `GpuBaseLimitExecIterator`? It would make it a little clearer.

Reply:
Updated. I changed the new object "GpuBaseLimitExec" to a class called `GpuBaseLimitIterator `.